synapseclient/core/upload/multipart_upload.py

```diff
@@ -480,7 +479,7 @@ def part_fn(part_number):
         return _get_file_chunk(file_path, part_number, part_size)
 
     def md5_fn(part, _):
-        md5 = hashlib.md5()
+        md5 = hashlib.new("md5", usedforsecurity=False)
```

Comment:
These changes around MD5 changes for the same reason as: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/976#discussion_r1244461825

In order to support python 3.8 `usedforsecurity` can only be defined in the `.new` call - Newer python versions let you to do `hashlib.md5(data, usedforsecurity=False)`

Reply:
Should we add a TODO here to update when py 3.8 is at EOL?

Reply:
It'll work in either way - We don't get a real benefit from one or the other